### PR TITLE
Fix bug 1395179 - Status numbers in progress menu should be clickable

### PR DESCRIPTION
**Pull request by katherine95**
#### What does this PR do?
This pull request fixes the issue [1395179](https://bugzilla.mozilla.org/show_bug.cgi?id=1395179). It ensures the status numbers in the progress menu are clickable.

#### Type of change:
- [x] Bug fix (non-breaking change which fixed an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update.

#### How Has This Been Tested?
- Clone this repository 
- Git checkout to `katherine95:1395179-clickable-status-numbers` branch.
- Follow the setup guidelines to set up pontoon locally
- Run `make run`
- Login to Pontoon
- Navigate to the progress menu and click the percentage.
- Click on the status numbers. For example, click  the number under "MISSING"

Expected result:

It navigates to `http://localhost:8000/translate/en-GB/tutorial/all-resources/?status=missing&string=1`

#### Checklist:
- [x] On click on any of the status numbers, it should navigate to the respective url.

#### Screenshot
Screenshot from 2019-11-13 14-50-35

### CODE REVIEW

File: frontend/src/modules/search/components/SearchBox.js [hunk 1]
```diff
@@ -80,12 +81,14 @@ export class SearchBoxBase extends React.Component<InternalProps, State> {
     updateFiltersFromURLParams = () => {
         const props = this.props;
 
-        const statuses = this.getInitialStatuses();
-        if (props.parameters.status) {
-            props.parameters.status.split(',').forEach(f => {
-                statuses[f] = true;
-            });
-        }
+        // let statuses = this.getInitialStatuses();
```
Comment (katherine95) [inline frontend/src/modules/search/components/SearchBox.js]: @adngdb can I remove this code? Because the status state is updated by `getDerivedStateFromProps`.
Or I just uncomment and leave it as it was?

Comment (adngdb) [reply frontend/src/modules/search/components/SearchBox.js]: Unless there's a good reason, it's better not to leave commented code around.

**Comment by adngdb** [comment]
The tests are failing, please look into it (and please remember to run them locally before sending your work ;-) ).

**Comment by katherine95** [comment]
> The tests are failing, please look into it (and please remember to run them locally before sending your work ;-) ).

I am using `make test-frontend` to run frontend tests and there are no failing tests. What could I be missing? @adngdb I just realized if I alter the file with the tests I would like to run is when it starts watching the file.

**Comment by adngdb** [comment]
@katherine95 By default, when you run `make test-frontend`, the test runner (`jest`) will look at your git status and only run tests for files that have changed. So if you commit your changes, then run the command, it won't run any tests. To change that, you can press `a` to force it to run all tests. You can press `w` to see the list of options in `jest`.

**Comment by katherine95** [comment]
I see thank you. Let me fix the failing tests

**Comment by katherine95** [comment]
> I can still reproduce the bug described in https://bugzilla.mozilla.org/show_bug.cgi?id=1395179#c4. Please make sure you have fixed the issue by testing it locally.

@adngdb do you mean when I click on the All resources drop-down, and click any of the `All` filters, nothing happens? On my side, everything works fine. Let me find an app to record a gif.

**Comment by adngdb** [comment]
It's weird if it works for you, because on my local copy of your branch, it doesn't. When I load the translate page for any project, and click the filter panel (next to the "Search" bar, top left of the screen) then click on any filter there, nothing happens. Are you sure you pushed all your commits?

**Comment by katherine95** [comment]
> It's weird if it works for you, because on my local copy of your branch, it doesn't. When I load the translate page for any project, and click the filter panel (next to the "Search" bar, top left of the screen) then click on any filter there, nothing happens. Are you sure you pushed all your commits?

Is this part that you mean? I haven't been working on it. Let me do so.
Screenshot from 2019-12-03 23-54-33

**Comment by katherine95** [comment]
I have been working on the progress menu part only, to ensure the status numbers are clickable and update the search box with the correct filter status. 
So I want to start working on the filters that you just brought to my attention. Thanks @adngdb

File: frontend/src/modules/search/components/SearchBox.js [hunk 2]
```diff
@@ -143,6 +144,30 @@ export class SearchBoxBase extends React.Component<InternalProps, State> {
         }
     }
 
+    static getDerivedStateFromProps(prevProps: InternalProps, state: Object) {
```
Comment (katherine95) [inline frontend/src/modules/search/components/SearchBox.js]: Without the `getDerivedStateFromProps` method, the filter panel works as it should, but on the flip side, the status value in the search box does not get updated when you click on the status numbers on the progress menu.
Might you have a suggestion on how I can go about this? @adngdb

**Comment by adngdb** [comment]
@katherine95 I figured it was easier to just send a commit to your branch, since I had to write the code to understand the problem anyway. The solution is here: https://github.com/mozilla/pontoon/pull/1498/commits/9d484a1beadf357ec38d748fd2f6f23be403af6c

@mathjazz mind giving this a quick try to make sure all works please?

File: frontend/yarn.lock [hunk 3]
```diff
@@ -6447,6 +6447,11 @@ lodash.isequal@4.5.0, lodash.isequal@^4.5.0:
   resolved "https://registry.yarnpkg.com/lodash.isequal/-/lodash.isequal-4.5.0.tgz#415c4478f2bcc30120c22ce10ed3226f7d3e18e0"
   integrity sha1-QVxEePK8wwEgwizhDtMib30+GOA=
 
+lodash.mapvalues@4.6.0:
+  version "4.6.0"
+  resolved "https://registry.yarnpkg.com/lodash.mapvalues/-/lodash.mapvalues-4.6.0.tgz#1bafa5005de9dd6f4f26668c30ca37230cc9689c"
+  integrity sha1-G6+lAF3p3W9PJmaMMMo3IwzJaJw=
+
```
Comment (mathjazz) [inline frontend/yarn.lock]: I assume we can revert this change?

Comment (adngdb) [reply frontend/yarn.lock]: Ah yes, thanks!

**Comment by adngdb** [comment]
Alrighty! Thanks a lot for your contribution @katherine95!
